### PR TITLE
[Merged by Bors] - feat: actions of `DomMulAct` on `A →[N] B` and `A →+[N] B`

### PR DESCRIPTION
**Pull request by urkud**
Refs #5379 

---

- [x] depends on: #5368
- [x] depends on: #5377 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5368
* leanprover-community/mathlib4#5377
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5368~~
* ~~leanprover-community/mathlib4#5377~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/GroupTheory/GroupAction/DomAct/ActionHom.lean [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+Copyright (c) 2023 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.GroupTheory.GroupAction.DomAct.Basic
+import Mathlib.Algebra.Hom.GroupAction
+/-!
+# Action of `Mᵈᵐᵃ` on `α →[N] β` and `A →+[N] B`
+
+In this file we define action of `DomMulAction M = Mᵈᵐᵃ` on `α →[N] β` and on `A →+[N] B`. At the
```
Comment (sgouezel) [inline Mathlib/GroupTheory/GroupAction/DomAct/ActionHom.lean:11]:
```suggestion
In this file we define action of `DomMulAct M = Mᵈᵐᵃ` on `α →[N] β` and on `A →+[N] B`. At the
```

File: Mathlib/GroupTheory/GroupAction/DomAct/ActionHom.lean [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+time of writing, these homomorphisms are not widely used in the library, so we put these instances
+into a separate file, not with the definition of `DomMulAction`.
```
Comment (sgouezel) [inline Mathlib/GroupTheory/GroupAction/DomAct/ActionHom.lean:13]:
```suggestion
into a separate file, not with the definition of `DomMulAct`.
```

File: Mathlib/GroupTheory/GroupAction/DomAct/ActionHom.lean [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    SMulCommClass Mᵈᵐᵃ M'ᵈᵐᵃ (A →+[N] B) :=
+  FunLike.coe_injective.smulCommClass (fun _ _ ↦ rfl) (fun _ _ ↦ rfl)
+
+theorem smul_mulDistribActionHom_apply (c : Mᵈᵐᵃ) (f : A →+[N] B) (a : A) :
+    (c • f) a = f (mk.symm c • a) :=
+  rfl
+
+@[simp]
+theorem mk_smul_mulDistribActionHom_apply (c : M) (f : A →[N] B) (a : A) :
+    (mk c • f) a = f (c • a) := rfl
```
Comment (sgouezel) [inline Mathlib/GroupTheory/GroupAction/DomAct/ActionHom.lean:71]: Why do you have a theorem on `A →[N] B` in a section on `A →+[N] B`? Is it the same as `mk_smul_mulActionHom_apply`?

File: Mathlib/GroupTheory/GroupAction/DomAct/ActionHom.lean [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+instance [DistribSMul M' A] [SMulCommClass M' N A] [SMulCommClass M M' A] :
+    SMulCommClass Mᵈᵐᵃ M'ᵈᵐᵃ (A →+[N] B) :=
+  FunLike.coe_injective.smulCommClass (fun _ _ ↦ rfl) (fun _ _ ↦ rfl)
+
+theorem smul_mulDistribActionHom_apply (c : Mᵈᵐᵃ) (f : A →+[N] B) (a : A) :
+    (c • f) a = f (mk.symm c • a) :=
+  rfl
+
+@[simp]
+theorem mk_smul_mulDistribActionHom_apply (c : M) (f : A →[N] B) (a : A) :
```
Comment (urkud) [inline Mathlib/GroupTheory/GroupAction/DomAct/ActionHom.lean:70]:
```suggestion
theorem mk_smul_mulDistribActionHom_apply (c : M) (f : A →+[N] B) (a : A) :
```

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**